lib/compiler/semantic.js

```diff
@@ -555,6 +578,28 @@ class SemanticPass {
         };
         this.scope = saved_scope;
     }
+    sa_NativeModuleDef(node) {
+        var code = '(function (juttle) {\n' +
+            'var module = {};\n' +
+            node.source + '\n' +
+            'return module.exports; })';
+        var module = eval(code)(runtime);
+        node.source = code;
+        var scope = new Scope(null, 'module');
+        node.elements = [];
+        node.uname = this.scope.alloc_var('module');
+        _.each(module.procs, (code, name) => {
+            this.enter_scope('sub');
+            var symbol = scope.define_native_proc(name, [], true);
+            symbol.module_uname = node.uname;
+            this.exit_scope();
```

Comment:
The body of the loop can be simplified to:

```js
scope.define_native_proc(name, [], true, node.uname);
```

with appropriately extended definition of `define_native_proc`.